### PR TITLE
Include lock funds in withdraw function

### PR DESCRIPTION
**Pull request by badb**

### CODE REVIEW

File: tests/golem/transactions/ethereum/test_ethereumtransactionsystem.py [hunk 1]
```diff
@@ -159,3 +159,30 @@ def test_withdraw(self, new_sci):
...
+                           1)
+        assert res == [gnt_tx, gntb_tx]
+        sci.transfer_gnt.assert_called_once_with(destination, gnt_balance)
+        sci.convert_gntb_to_gnt.assert_called_once_with(
+            destination,
+            gntb_balance - 1,
+        )
+        sci.reset_mock()
+
+        # Enough total GNT
```
Comment (maaktweluit) [inline tests/golem/transactions/ethereum/test_ethereumtransactionsystem.py]: This comment seems incorrect, `not` is missing?

**Review by maaktweluit — CHANGES_REQUESTED** [review]
LGTM!

One small comment question

File: golem/client.py [hunk 2]
```diff
@@ -865,8 +865,14 @@ def withdraw(
...
         if isinstance(amount, str):
             amount = int(amount)
+        gnt_lock, eth_lock = self.funds_locker.sum_locks()
+        if currency == 'GNT':
+            lock = gnt_lock
+        else:
+            lock = eth_lock
 
-        return self.transaction_system.withdraw(amount, destination, currency)
+        return self.transaction_system.withdraw(amount, destination, currency,
```
Comment (Krigpl) [inline golem/client.py:874]: just do `amount-lock`, no need for `transaction_system.withdraw` to be aware of any "locks"

**Review by Krigpl — CHANGES_REQUESTED** [review]

File: tests/golem/transactions/ethereum/test_ethereumtransactionsystem.py (hunk 1 above)
Comment (Krigpl) [reply tests/golem/transactions/ethereum/test_ethereumtransactionsystem.py]: yesquiteindeed

File: golem/client.py (hunk 2 above)
Comment (jiivan) [reply golem/client.py:874]: @Krigpl did you mean `total-lock`?

Comment (Krigpl) [reply golem/client.py:874]: Oh yeah, I realized my comment doesn't actually make any sense. The current approach is correct.

**Review by Krigpl — APPROVED** [review]

**Review by jiivan — APPROVED** [review]

**Pull request merged**